### PR TITLE
Justin krs/user segments

### PR DESCRIPTION
**Pull request by justinengle-krs**
Add user segments . Create generic control that an be used on both user and customer segments
Zeplin:
image
Implementation:
image

### CODE REVIEW

**Review by casperlotter — DISMISSED** [review]

File: src/components/Segments/AddSegmentModal.svelte [hunk 1]
```diff
@@ -44,18 +45,12 @@
   <div slot="content">
     <div class="px-8 py-8 border-b border-neutral-200">
       <!--todo :dynamic when used on user page-->
-      <p class="text-neutral-500 text-regular font-bold mb-4">Only show this customer's products where the...</p>
+      <p class="text-neutral-500 text-regular font-bold mb-4">
+        {segmentContext == 'customers' ? "Only show this customer's products where the..." : 'Users can only see...'}
```
Comment (matthampton-stock2shop) [inline src/components/Segments/AddSegmentModal.svelte:49]: Could you tweak this a little:

"Only show this customer's products where the..." -> "Customer  can only see products where the..."

'Users can only see...' -> 'User can only see...'

**Review by matthampton-stock2shop — APPROVED** [review]

File: src/components/Segments/AddSegmentModal.svelte (hunk 1 above)
Comment (justinengle-krs) [reply src/components/Segments/AddSegmentModal.svelte]: Done @matthampton-stock2shop

Comment (justinengle-krs) [reply src/components/Segments/AddSegmentModal.svelte]: https://www.chromatic.com/build?appId=63ef6cddab1dec384359e1cd&number=78

**Comment by matthampton-stock2shop** [comment]
https://www.chromatic.com/build?appId=63ef6cddab1dec384359e1cd&number=80

**Review by matthampton-stock2shop — APPROVED** [review]

**Pull request merged**